ibis/expr/sql.py

```diff
@@ -276,6 +374,24 @@ def convert_in(in_, catalog):
     return this.isin(candidates)
 
 
+@convert.register(sge.Cast)
+def cast(cast, catalog):
+    this = convert(cast.this, catalog)
+    to = convert(cast.to, catalog)
+
+    return this.cast(to)
+
+
+@convert.register(sge.DataType)
+def datatype(datatype, catalog):
+    return SqlglotType().to_ibis(datatype)
+
+
+@convert.register(sge.Count)
+def count(count, catalog):
+    return ibis._.count()
```

Comment:
Since `Count` can operate on columns, and a count on a column has a different meaning than on a table (the column version ignores nulls like other aggregates), this can't just be underscore-dot-count.

This is what is causing some of the test failures in CI.